### PR TITLE
TCVP-1211 Added staff-api update dispute endpoint

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-1211](https://justice.gov.bc.ca/jira/browse/TCVP-1211)
- Added functionality to update a specified dispute's fields based on its id.
- Added staff-api wrapper for update dispute endpoint. 
- Updated oracle-data-api dispute model to make some fields nullable in the open api spec.
- Added junit and xunit tests.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/TicketCount.java [hunk 1]
```diff
@@ -32,5 +36,9 @@ public class TicketCount {
 
 	@Column
 	private boolean fineReductionRequest;
+	
+	@JsonBackReference
+	@ManyToOne(targetEntity=Dispute.class, fetch = FetchType.LAZY)
+	private Dispute dispute;
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/TicketCount.java:42]: Is there a way to hide this property from the generated swagger def?  If I exclude the "id"  and "dispute" properties from a Postman POST call to /api/v1.0/dispute, I'm able to create a Dispute just fine.
```
  "ticketCounts": [
    {
      "offenceDeclaration": "string",
      "timeToPayRequest": true,
      "fineReductionRequest": true
    },
    {
      "offenceDeclaration": "string",
      "timeToPayRequest": true,
      "fineReductionRequest": true
    }
  ],
```

**Review by KDNeufeld — APPROVED** [review]
Great work!  LGTM.
Oh, BTW, now that redis is part of staff-api, don't forget to add necessary env vars to docker-compose for this.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java [hunk 2]
```diff
@@ -141,6 +141,24 @@ public void cancelDispute(@PathVariable Integer id) {
...
+	 * @param dispute to be updated
+	 * @param id of the saved {@link Dispute} to update
+	 * @return updated {@link Dispute}
+	 */
+	@Operation(summary = "Updates the properties of a particular Dispute record based on the given values.")
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "Ok"),
+		@ApiResponse(responseCode = "400", description = "Bad Request."),
+		@ApiResponse(responseCode = "404", description = "Dispute record not found. Update failed.")
+	})
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:157]: I think 405 should also be returned if there are validation errors in the submitted Dispute.  Ie, the rejectedReason can only be 256 characters long.  Currently if I try to update a Dispute and put a rejectedReason with > 256 this method returns a 500 error, not 405 error.

Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:157]: We can add this later in a separate PR.  I noticed the POST also doesn't do this but should.

Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:157]: That's true. Yes, maybe we can create a separate PR for this.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeService.java [hunk 3]
```diff
@@ -56,6 +56,44 @@ public void save(Dispute dispute) {
...
+			disputeToUpdate.getTicketCounts().clear();
+		}
+		// Add updated ticket counts
+		disputeToUpdate.addTicketCounts(dispute.getTicketCounts());
+		disputeToUpdate.setLawyerRepresentation(dispute.isLawyerRepresentation());
+		disputeToUpdate.setInterpreterLanguage(dispute.getInterpreterLanguage());
+		disputeToUpdate.setWitnessIntent(dispute.isWitnessIntent());
+		
+		return disputeRepository.save(disputeToUpdate);
+	}
```
[the remark this answers is not in the document]
Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeService.java:96]: Good call. Will update this.

**Pull request merged**